### PR TITLE
Feature/adf ci cd auto publish

### PR DESCRIPTION
**Pull request by manjitsin**
# Type of PR

- Documentation changes
- Code changes
- CI-CD changes

## Purpose

- The Purpose of this PR is to- Create a sample CI/CD For Azure Data Factory that can eliminate the Manual Publish and allow Auto Publish of ARM Templates that can be used for a fully automated E2E deployment on ADF in multiple environments.

## Does this introduce a breaking change? If yes, details on what can break
NO
## Author pre-publish checklist

- [X] Added test to prove my fix is effective or new feature works
- [X] No PII in logs
- [X] Made corresponding changes to the documentation

## Validation steps
- Follow steps in README

## Issues Closed or Referenced

- References [#issue_number](https://github.com/cse-labs/code-with-dataops/issues/422)

### CODE REVIEW

**Comment by promisinganuj** [comment]
Hi @manjitsin, thanks for the contribution.

While reviewing the PR, I see that it has quite a lot of overlap with the following Microsoft documentation on the same topic https://learn.microsoft.com/en-us/azure/data-factory/continuous-integration-delivery-improvements.

I would request you to review the above document and identify areas where we can add value. One idea that comes to my mind is that we can provide a deployable solutions to showcase this functionality i.e. the sample would have a `deploy.sh` or something which can spin-up an ADF and Azure DevOps environment.

Please feel free to ping me for further discussion.

File: single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md [hunk 1]
```diff
@@ -0,0 +1,322 @@
+# Automated Publish and CI/CD for Azure Data Factory
+
+This sample demonstrates the deployment of Azure Data Factory using Automated Publish Method. Usually, the Azure Data Factory deployment required Manual Publish setup where the developer had to manually Publish the ADF changes from the portal. This step generated the ARM Templates which were used in deployment step.
```
Comment (promisinganuj) [inline single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:3]: Automated Publish Method => `Automated Publish` feature of CI/CD.

Comment (promisinganuj) [reply single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:3]: required Manual Publish setup => requires `Manual Publish` setup

Comment (promisinganuj) [reply single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:3]: developer had to manually Publish the ADF changes => developer publishes the ADF changes manually

Comment (promisinganuj) [reply single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:3]: This step generated the ARM Templates which were used in deployment step. => This step generates ARM Templates which are used in deployment steps.

File: single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+In this sample, we have eliminated the Manual Publish step by using ADF Packages for Automated Publish. Now, the ARM templates will be generated automatically without any manual intervention.
```
Comment (promisinganuj) [inline single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:5]: In this sample, the need to manually publish step has been eliminated

Comment (promisinganuj) [reply single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:5]: ADF Packages for Automated Publish. => by using a publicly available npm package [@microsoft/azure-data-factory-utilities](https://www.npmjs.com/package/@microsoft/azure-data-factory-utilities) for automated publishing.

Comment (promisinganuj) [reply single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:5]: Now, the ARM templates will be generated automatically without any manual intervention. => With this, the ARM templates can be generated automatically without any manual intervention.

File: single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+
+## ADF Introduction
+
+Azure Data Factory(ADF) is a service in Microsoft Azure that allows the developers to integrate and transform their data using various processes. ADF has good integrations with multiple internal and external services.
```
Comment (promisinganuj) [inline single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:9]: Azure Data Factory(ADF) => Azure Data Factory (ADF)

Comment (promisinganuj) [reply single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:9]: Azure Data Factory is a cloud-based, fully managed, serverless data integration service for data movement and transformation. It enables users to create, schedule, and orchestrate data pipelines that move data between various on-premises and cloud-based data sources to a central data store. The pipelines can be used to perform complex data transformations, including data enrichment, data validation, and data migration.

File: single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+
+In ADF, CI/CD essentially means deploying the Data Factory components(Linked Service, Triggers, Pipelines, Datasets) to a new environment(ADF) automatically using Pipelines.
```
Comment (promisinganuj) [inline single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:11]: the Data Factory components(Linked Service, Triggers, Pipelines, Datasets) => various data factory entities such as linked service, triggers, pipelines, datasets etc.

Comment (promisinganuj) [reply single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:11]: new environment(ADF) => new environment

Comment (promisinganuj) [reply single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:11]: Pipelines => Azure Pipelines (you can also add link to it https://learn.microsoft.com/en-us/azure/devops/pipelines/get-started/what-is-azure-pipelines?view=azure-devops)

File: single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+
+### ADF Modes
+
+1) ADF consists of two modes- Live Mode and Git Mode.
```
Comment (promisinganuj) [inline single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:15]: 1) ADF consists of two modes- Live Mode and Git Mode.
=>
ADF consists of two modes namely `Live Mode` and `Git Mode`.

Comment (promisinganuj) [reply single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md:15]: Add two numbered bullets to explain these two different modes in few words:

1. Live Mode is ...
2. Git Mode - ...

**Review by promisinganuj — CHANGES_REQUESTED** [review]
To be continued...

File: single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md (hunk 4 above)
Comment (manjitsin) [reply single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md]: DOne

File: single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md (hunk 5 above)
Comment (manjitsin) [reply single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md]: Done

File: single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md (hunk 2 above)
Comment (manjitsin) [reply single_tech_samples/datafactory/sample2_cicd_auto_publish/README.md]: Done

**Review by promisinganuj — APPROVED** [review]
LGTM!

**Pull request merged**